punctfix/streaming.py

```diff
@@ -0,0 +1,124 @@
+from typing import List, Optional
+
+from punctfix.inference import PunctFixer, WordPrediction
+
+
+class PunctFixStreamer:
+    """
+    A stateful streamer that receives text in segments, on-line performing punct-fixing and
+    returning partial results during streaming. These partial results are guaranteed to be
+    final.
+    """
+
+    chunked_words: List[WordPrediction]
+    buffer: List[WordPrediction]
+
+    def __init__(self, punct_fixer: PunctFixer):
+        """
+        Takes in an instantiated punct fixer.
+        """
+        self.punct_fixer = punct_fixer
+        self.clear()
+
+    def __call__(self, new_text_segment: str) -> Optional[str]:
+        """
+        Stream in new text, returning None if this new text did not change anything
+        and the partial, finalized text if there has been updates to it.
+        """
+        self.buffer.extend(
+            self.punct_fixer.init_word_prediction_list(
```

Comment:
As long as one entire text, (+ some storage for labels) so it would not use any more memory than normal punctfixer, it just keeps the memory for longer